File: clockDemo/src/main/java/com/dx/demi/activity/MainActivity.java
package com.dx.demi.activity;

import android.content.Intent;
import android.content.res.ObbInfo;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.dx.demi.R;

import java.sql.Array;
import java.util.HashMap;
import java.util.Iterator;
import java.util.Set;

import static android.support.v7.widget.RecyclerView.*;


public class MainActivity extends AppCompatActivity {

    RecyclerView rcv ;
    Class[] clazz = {PieChatActivity.class, ClockActivity.class,CoordinateActivity.class,EventActivity.class,
            ListActivity.class,NotifyCationActivity.class,ShopCarActivity.class,CameraActivity.class,
            EToastActivity.class,TimeConutDownActivity.class,AnimationActivity.class,ViewPagerActivity.class,
            DownloadActivity.class,RetrofitOKHttpActivity.class};
    String[] describle = {"水果大拼盘","时钟表盘","坐标系","事件监听",
            "多布局RecycleView列表","通知栏事件监听","购物车动画","照相机",
    "自定义Toast","倒计时","动画","ViewPager","下载apk","Retrofit"};
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        rcv =(RecyclerView)findViewById(R.id.rcv);
        rcv.setLayoutManager(new LinearLayoutManager(this));
        rcv.setAdapter(new MyRecyclerAdapter());
       // rcv.addItemDecoration(new DividerItemDecoration(this, DividerItemDecoration.HORIZONTAL_LIST)));
    }

    class MyRecyclerAdapter extends RecyclerView.Adapter<MyRecyclerAdapter.MyViewHolder>{
        @Override
        public MyViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
            MyViewHolder viewHolder =
                    new MyViewHolder(LayoutInflater.from(MainActivity.this).inflate(
                            R.layout.rev_item,parent,false));
            return viewHolder;
        }

        @Override
        public void onBindViewHolder(MyViewHolder holder, final int position) {
            holder.text_content.setText(describle[position]);
            holder.text_content.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    MainActivity.this.startActivity(new Intent(MainActivity.this,clazz[position]));
                }
            });
        }

        @Override
        public int getItemCount() {
            return clazz.length;
        }
        /**
         *印证上面所说的，这边定义的内部类MyViewHolder需要继承RecyclerView.ViewHolder，
         *这样才能为系统所重用。
         **/
        class MyViewHolder extends ViewHolder{
            public TextView text_content;
            public MyViewHolder(View itemView) {
                super(itemView);
                /**
                 *初始化item里面的控件
                 */
                text_content = (TextView) itemView.findViewById(R.id.text_content);
            }
        }
    }

}

File: app/src/main/java/com/xd/demi/activity/MainActivity.kt
package com.xd.demi.activity

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import com.chad.library.adapter.base.BaseQuickAdapter
import com.xd.demi.R
import com.xd.demi.adapter.DividerItemDecoration
import com.xd.demi.adapter.SamplesAdapter
import com.xd.demi.bean.Samples
import kotlinx.android.synthetic.main.activity_main.*
import java.util.*


class MainActivity : Activity() {
    private var clazz = arrayOf(TestActivity::class.java, CustomViewActivity::class.java, ProduceConsumerActivity::class.java, ScreenMatchActivity::class.java, FirstActivity::class.java, MyServiceActivity::class.java, MatrixActivity::class.java, CoordinateActivity::class.java, EventActivity::class.java, ListActivity::class.java, NotifyCationActivity::class.java, EToastActivity::class.java, AnimationActivity::class.java, DownloadActivity::class.java, RetrofitOKHttpActivity::class.java, RxJavaActivity::class.java, PortfoliosActivity::class.java, VideoPlayerActivity::class.java)
    private var describle = arrayOf("临时测试Activity", "自定义View", "ProducerConsumer", "屏幕适配", "Activity生命周期", "MyServiceActivity", "Matrix", "坐标系", "事件监听", "多布局RecycleView列表", "通知栏事件监听", "自定义Toast", "动画", "下载apk", "Retrofit", "RXJava", "收益走势图", "视频播放器")
    private var sampleList = ArrayList<Samples>()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        for (i in describle.indices) {
            val mSample = Samples()
            mSample.description = describle[i]
            mSample.mClass = clazz[i]
            sampleList.add(mSample)
        }
        val sampleAdapter = SamplesAdapter(sampleList, this)
        sampleAdapter.openLoadAnimation()
        rcv.layoutManager = LinearLayoutManager(this)
        rcv.adapter = sampleAdapter
        rcv.addItemDecoration(DividerItemDecoration(this))
        sampleAdapter.onItemChildClickListener = BaseQuickAdapter.OnItemChildClickListener { adapter, _, position ->
            val sample = adapter.getItem(position) as Samples
            startActivity(Intent(this@MainActivity, sample.mClass))
        }
    }

    fun add(a: Int, b: Int): Int {
        return a + b
    }

}
